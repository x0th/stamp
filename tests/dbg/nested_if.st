Dog = Object^;
Dog barks = true;
Dog meows = false;

Cat = Object^;

if (Dog.barks) {
	if (Dog.meows) {
		Cat.type
	} else {
		Dog.type
	}
}
