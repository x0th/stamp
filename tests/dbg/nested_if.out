STDOUT:
----------------------
Context:
Callable = Callable-hash
List = List-hash
Object = Object-hash
false = false
if = if
true = true
----------------------
TokSList()
  TokSend()
    TokObject(Object)
    TokMessage(clone)
      TokObject(Dog)
  TokStore()
    TokObject(Dog)
    TokValue(barks)
    TokObject(true)
  TokStore()
    TokObject(Dog)
    TokValue(meows)
    TokObject(false)
  TokSend()
    TokObject(Object)
    TokMessage(clone)
      TokObject(Cat)
  TokSend()
    TokObject(if)
    TokMessage(if_true)
      TokSList()
        TokSend()
          TokObject(Dog)
          TokMessage(barks)
        TokSList()
          TokSList()
            TokSend()
              TokObject(if)
              TokMessage(if_true)
                TokSList()
                  TokSend()
                    TokObject(Dog)
                    TokMessage(meows)
                  TokSList()
                    TokSList()
                      TokSend()
                        TokObject(Cat)
                        TokMessage(type)
                  TokSList()
                    TokSend()
                      TokObject(Dog)
                      TokMessage(type)
        TokSList()
Dog
----------------------
Context:
Callable = Callable-hash
Cat = Cat-hash
Dog = Dog-hash
List = List-hash
Object = Object-hash
false = false
if = if
true = true
----------------------
STDERR:
